node_modules/col-activities/lib/notifications/daily.js

```diff
@@ -0,0 +1,548 @@
+ * @param  {Function}   callback      Standard callback function
+ * @api private
+ */
+var handleUser = function(course, data, user, callback) {
+  // Ignore users that don't have an email address
+  if (!user.canvas_email) {
+    return callback();
+  }
+
+  // Get all the activities for the user
+  var activities = getActivitiesForUser(data, user);
+
+  // If nothing noteworthy happened for this user, then they should not get an email
+  if (_.isEmpty(activities)) {
+    return callback();
+  }
+
+  var subject = getSubject(course, activities, user);
+
+  // The data that's used in the daily email template
+  var data = {
+    'activities': activities,
+    'getSummaryActors': getSummaryActors,
+  };
+  EmailUtil.sendEmail(subject, user, course, data, 'daily', function(err) {
+    // Don't pass errors up as otherwise all the following users would be ignored.
+    // The error itself is logged in the sendEmail implementation
+    return callback();
+  });
+};
+
+/**
+ * Get the subject for the daily email
+ *
+ * @param  {Course}                                 course        The course for which to send out the daily notifications
+ * @param  {(AssetActivity|WhiteboardActivity)[]}   activities    The set of activities for which to generate an email subject
+ * @param  {User}                                   currentUser   The user to whom the email will be sent
+ * @return {String}                                               An appropriate email subject
+ * @api private
+ */
+var getSubject = function(course, activities, currentUser) {
+  var subject = null;
+  if (activities.length === 1) {
+    var activity = activities[0];
+    if (activity.type === 'asset_comment') {
+      var commenters = _.pluck(activity.asset.comments, 'user');
+      var summaryActors = getSummaryActors(commenters, currentUser);
+      subject = summaryActors + ' commented on your asset ‘' + activity.asset.title + '’';
+    } else if (activity.type === 'asset_comment_reply') {
+      var commenters = _.pluck(activity.asset.comments, 'user');
+      var summaryActors = getSummaryActors(commenters, currentUser);
+      var comment = activity.asset.comments[0].body;
+      if (comment.length > 40) {
+        comment = comment.substring(0, 40) + '...';
+      }
+      subject = summaryActors + ' replied to your comment ‘' + comment + '’';
+    } else if (activity.type === 'whiteboard_chat') {
+      var commenters = _.pluck(activity.whiteboard.chats, 'user');
+      var summaryActors = getSummaryActors(commenters, currentUser);
+      subject = summaryActors + ' commented on your whiteboard ‘' + activity.whiteboard.title + '’';
+    }
+
+  // Default to a generic subject title if there's more than 1 activity
+  // If the canvas instance for this course has no name, we change the subject slightly as the
+  // "from name" already contains "Asset Library and Whiteboards"
+  } else {
+    subject = 'Today\'s activity in the Asset Library and Whiteboards';
```

Comment:
This can be changed to:

- Both Asset Library and Whiteboards activity: `New Asset Library and Whiteboard activity is waiting for you`
- Only Asset Library activity: `New Asset Library activity is waiting for you`
- Only Whiteboards activity: `New Whiteboard activity is waiting for you`
- No Canvas name:  `New activity is waiting for you`